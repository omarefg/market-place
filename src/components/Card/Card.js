import React from "react";
import classes from './Card.module.css'

export function Card({ img, alt, caption, href, onAddToCart }) {
  return (
    <figure className={classes.root}>
      <img className={classes.cardImg} src={img} alt={alt} />
      <figcaption className={classes.caption}>{caption}</figcaption>
      <button onClick={onAddToCart}>Add to cart</button>
    </figure>
  );
}
