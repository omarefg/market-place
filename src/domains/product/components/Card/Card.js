import React from "react";
import classes from './Card.module.css'

export function Card(props) {
  const { img, alt, caption, onAddToCart, type, id } = props

  const handleOnAddToCart = () => {
    onAddToCart({ type, id })
  }

  return (
    <figure className={classes.root}>
      <img className={classes.cardImg} src={img} alt={alt} />
      <figcaption className={classes.caption}>{caption}</figcaption>
      <button onClick={handleOnAddToCart}>Add to cart</button>
    </figure>
  );
}
