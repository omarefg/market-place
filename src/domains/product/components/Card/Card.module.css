.root {
  margin: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  user-select: none;
  background-color: var(--color-2);
  border-radius: 8px;
  padding: 16px;
}

.caption {
  text-align: center;
  margin-bottom: 10px;
  flex: 1;
  color: var(--color-1);
}

.cardImg {
    object-fit: contain;
    width: 100%;
    border-radius: 8px;
  }
