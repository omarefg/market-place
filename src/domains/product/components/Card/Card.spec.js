import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Card } from "./Card";

describe("@domains/product/components/Card", () => {
  let defaultProps, wrapper;

  beforeEach(() => {
    defaultProps = {
      img: "imageMock",
      alt: "altMock",
      caption: "captionMock",
      onAddToCart: jest.fn(),
      type: "typeMock",
      id: "idMock",
    };

    wrapper = (props = {}) => {
      const finalProps = { ...defaultProps, ...props };
      return render(<Card {...finalProps} />);
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Given a valid call it must render without crashing", () => {
    // Arrange
    const { container } = wrapper();
    // Assert
    expect(container).toMatchSnapshot();
  });

  it("Given a click event inside add button it must call onAddToCart prop", () => {
    // Arrange
    const { getByTestId } = wrapper();
    const button = getByTestId("Card__addButton");
    // Act
    fireEvent.click(button);
    // Assert
    expect(defaultProps.onAddToCart).toBeCalledTimes(1);
  });

  it("Given a click event inside add button it must call onAddToCartProp with type and id information", () => {
    // Arrange
    const { getByTestId } = wrapper();
    const button = getByTestId("Card__addButton");
    // Act
    fireEvent.click(button);
    // Assert
    expect(defaultProps.onAddToCart).toBeCalledWith({
      type: defaultProps.type,
      id: defaultProps.id,
    });
  });
});
