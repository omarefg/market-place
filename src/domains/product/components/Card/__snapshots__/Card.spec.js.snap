// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/product/components/Card Given a valid call it must render without crashing 1`] = `
<div>
  <figure
    class="root"
  >
    <img
      alt="altMock"
      class="cardImg"
      src="imageMock"
    />
    <figcaption
      class="caption"
    >
      captionMock
    </figcaption>
    <button
      data-testid="Card__addButton"
    >
      Add to cart
    </button>
  </figure>
</div>
`;
